ui/pages/onboarding-flow/onboarding-app-header/index.scss

```diff
@@ -2,10 +2,10 @@
 
 .onboarding-app-header {
   position: relative;
-  z-index: design-system.$header-z-index;
```

Comment:
header is suppose to have higher index than other elements, but lower than modal/overlay.. have you checked the index of overlays? i think the fix is to have the $header-z-index to have a lower index than the overlay index..